### PR TITLE
:sparkles: You can now enable delta bundles

### LINKED ISSUE #323: 🔧 Enable delta bundles
Enable the Delta Bundles

### PR DESCRIPTION
**Pull request by bdumpp**
Set default for delta bundles with `enableDeltaBundlesDefault` in the Controller configuration file, and set delta bundles for a specific system with `enableDeltaBundles` in System resource file.

Resolves #323

### CODE REVIEW

File: internal/controller/styra/system_controller.go [hunk 1]
```diff
@@ -1055,8 +1060,12 @@ func (r *SystemReconciler) updateSystem(
 	system *v1beta1.System,
 ) (*styra.SystemConfig, error) {
 	log.Info("Updating system")
-
 	cfg := r.specToSystemConfig(system)
+	log.Info(fmt.Sprintf("cfg: %v, system: %v, controller: %v",
```
Comment (bdjgs) [inline internal/controller/styra/system_controller.go:1064]: This seems like debug logging. Remove debug logging.

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -4,5 +4,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: controller
-  newTag: latest
+  newName: controller-config-db
```
Comment (bdjgs) [inline config/manager/kustomization.yaml:7]: revert

File: internal/controller/styra/system_controller.go [hunk 3]
```diff
@@ -1090,6 +1099,19 @@ func (r *SystemReconciler) specToSystemConfig(system *v1beta1.System) *styra.Sys
...
+	}
+	if system.Spec.EnableDeltaBundles != nil {
+		enableDeltaBundles = *system.Spec.EnableDeltaBundles
+	}
+
+	bundleDownloadConfig := &styra.BundleDownloadConfig{
+		DeltaBundles: enableDeltaBundles,
+	}
+	cfg.BundleDownload = bundleDownloadConfig
+
```
Comment (bdjgs) [inline internal/controller/styra/system_controller.go:1114]: In my opinion there is not reason to create a variable that is only used once. 

```suggestion
	cfg.BundleDownload = &styra.BundleDownloadConfig{
		DeltaBundles: enableDeltaBundles,
	}
```

File: api/config/v2alpha2/projectconfig_types.go [hunk 4]
```diff
@@ -38,9 +38,12 @@ type ProjectConfig struct {
 	// protection if it is not set on the resource.
 	DeletionProtectionDefault bool `json:"deletionProtectionDefault"`
 
-	// ReadOnly sets the default values of ReadOnly for systems
+	// ReadOnly sets the value of ReadOnly for systems
 	ReadOnly bool `json:"readOnly"`
 
+	// EnableDeltaBundlesDefault sets the default of whether systems have delta-bundles or not
+	EnableDeltaBundlesDefault *bool `json:"enableDeltaBundlesDefault,omitempty"`
```
Comment (bdjgs) [inline api/config/v2alpha2/projectconfig_types.go:45]: Is there a reason why it is a pointer to a bool and not just a bool?

Comment (bdumpp) [reply api/config/v2alpha2/projectconfig_types.go:45]: Ellers kan man ikke distinguish false fra unset

**Review by bdjgs — APPROVED** [review]

**Pull request merged**